### PR TITLE
DHFPROD-7830: popover

### PR DESCRIPTION
**Pull request by MSBrignoni**
### Description

Bootstrap Popover

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by briantang — CHANGES_REQUESTED** [review]
@MSBrignoni Looks like there is a popover related e2e failure. Can we also fix the commit message to have a colon after the ticket number? Should be "DHFPROD-7830: Popover"

Screen Shot 2021-11-29 at 10 44 27 AM

**Comment by MSBrignoni** [comment]
> @MSBrignoni Looks like there is a popover related e2e failure. Can we also fix the commit message to have a colon after the ticket number? Should be "DHFPROD-7830: Popover"
> 
> Screen Shot 2021-11-29 at 10 44 27 AM

Thank you Brian, done. Yes, i was fixing this in the "Authorization.spec".

**Review by briantang — CHANGES_REQUESTED** [review]
@MSBrignoni Looks good, but found a couple places where I think the popover needs to persist, but it destroys too early. In the video, we need to be able to click inside the popover, so the popover should not close until we blur out of the popover.
(1 in create-edit-step.tsx, 2 in entity-map-table.tsx (the stacked icons))

https://watch.screencastify.com/v/zW9F1mlEj8NbJyxLs7Eq

**Comment by MSBrignoni** [comment]
> @MSBrignoni Looks good, but found a couple places where I think the popover needs to persist, but it destroys too early. In the video, we need to be able to click inside the popover, so the popover should not close until we blur out of the popover. (1 in create-edit-step.tsx, 2 in entity-map-table.tsx (the stacked icons))
> 
> https://watch.screencastify.com/v/zW9F1mlEj8NbJyxLs7Eq

Yes, i see @briantang, thank you. What i did is, give it a delay of 2 seconds in the popover wich need interaction (only when it needs to hide), because this element of bootstrap doesn't controll the events like ant with the hover.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx [hunk 1]
```diff
@@ -1381,23 +1378,27 @@ const EntityMapTable: React.FC<Props> = (props) => {
...
         trigger="hover"
         placement="top"
-        getPopupContainer={() => document.getElementById("parentContainer") || document.body}><img className={styles.arrayImage} src={DocIcon} alt={""} data-testid="XPathInfoIcon" /></Popover>
+        delay={{show: 0, hide: 2000}}
+        rootClose
+      >
+        <img className={styles.arrayImage} src={DocIcon} alt={""} data-testid="XPathInfoIcon" />
+      </OverlayTrigger>
       </span>,
+
```
Comment (federicoRufrancosMarklogic) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx:1401]: here this whitespace has some logic? Or need remove it

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx [hunk 2]
```diff
@@ -1336,15 +1336,12 @@ const EntityMapTable: React.FC<Props> = (props) => {
             <div className={row.joinPropertyName !== "" || !expanded ? styles.renderContainer : styles.noKeyContainer}>
               {expanded && row.joinPropertyName !== "" ?
                 //if multiple and has foreign key, show context help
-                <div className={styles.typeContainer}>
+                <div className={styles.typeContainer} >
```
Comment (federicoRufrancosMarklogic) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx:1339]: extra space to the end of the tag

**Review by federicoRufrancosMarklogic — APPROVED** [review]
I left 2 minor comments, anyway it's approved

**Review by yubiselmarklogic — APPROVED** [review]

**Review by mrodrigu-marklogic — APPROVED** [review]

**Review by iveyMarklogic — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Pull request merged**